mvp-spec/partials-updater/src/index.ts

```diff
@@ -44,3 +46,20 @@ async function updateModifiedMermaids() {
     const modifiedMermaidFiles = mermaidFiles.filter(f => changedFiles.includes(f))
     await updateMermaids(modifiedMermaidFiles)
 }
+
+async function checkUnusedPartials() {
+    const partials = await listPartials()
+    const docs = await listDocuments()
+    let usedPartials = new Array<string>()
```

Comment:
NIT: prefer `const`, simplify intialization, use `push` to avoid re-affecting the array at each loop

```javascript
const usedPartials = [];
for (const doc of docs) {
    const document = await loadDocument(doc)
    const tokens = lex(document)
    const partialsFromToken = await listUsedPartials(tokens)
    usedPartials.push(partialsFromToken)
}
```

Reply:
Note that `partialsFromToken` is an array.

I wanted to avoid twisting with variadic parameters for simplicity. But in the end, I don't think we care. Changed.

BTW, avoiding things like `const usedPartials = [];`. It is dangerous due to misleading type inference. Particularly in this case...